packages/quantic/force-app/main/default/lwc/quanticResultLink/quanticResultLink.js

```diff
@@ -90,10 +99,47 @@ export default class QuanticResultLink extends LightningElement {
     );
   };
 
+  checkResultType() {
+    if (this.result?.raw?.sfid !== undefined) {
+      this.isSalesforceLink = true;
+    }
+  }
+
+  handleOpenTabSalesforceLink(event) {
+    event.stopPropagation();
+
+    const targetPageRef = {
+      type: 'standard__recordPage',
+      attributes: {
+        recordId: this.getRecordIdAttribute(),
+        objectApiName: this.getObjectApiNameAttribute(),
+        actionName: 'view',
+      },
+    };
+    this[NavigationMixin.Navigate](targetPageRef);
+  }
+
+  getRecordIdAttribute() {
+    let idToUse = this.result.raw.sfid;
+
+    // Knowledge article uses the knowledge article version id to navigate.
+    if (
+      this.result.raw.sfkbid !== undefined &&
+      this.result.raw.sfkavid !== undefined
+    ) {
+      idToUse = this.result.raw.sfkavid;
+    }
+    return idToUse || '';
+  }
+
+  getObjectApiNameAttribute() {
+    return this.result?.raw?.objecttype;
+  }
```

Comment:
Like Nathan already mentioned, a getter would be nice to have here as well:

```suggestion
  get objectType() {
    return this.result?.raw?.objecttype;
  }
```

Reply:
as brought up by @erocheleau I tested the NavigationMixin call without the objectApiName and it works well, so we dont need this at all in the end